pkg/streamingpromql/operators/functions/histograms.go

```diff
@@ -136,7 +140,7 @@ func (h *HistogramFunctionOverInstantVector) SeriesMetadata(ctx context.Context)
 		if !groupExists {
 			g.labels = series.Labels
 			g.group = bucketGroupPool.Get()
-			g.group.firstInputSeriesIdx = innerIdx
+			g.group.lastInputSeriesIdx = innerIdx
```

Comment:
This isn't quite right - we need to update `lastInputSeriesIdx` both when the group already exists and when it doesn't, but it's currently only set when the group is created for the first time.

Same comment applies for classic histogram case below.

Might be good to add some tests for the sorting logic to catch stuff like this.